lib/LLVMIRCodeGen/libobj/Readme.md

```diff
@@ -0,0 +1,19 @@
+# Archiving additional objects within the bundle object
+
+All the object files (.o) stored in this directory will be automatically
```

Comment:
Could you also provide the motivation for this feature and why it is useful?

Reply:
I added some a description. I paste it here:

This is useful for the LLVM based backends in the following scenario:
- A LLVM backend generates a bundle with external calls to functions which are not part
  of LIBJIT. For example this is the case for some kernel libraries which are pre-compiled
  with other compilers than Clang (the one used by default by Glow) for better performance.
- This means that the bundle generated by Glow is not self contained and requires the
  respective library/objects to be provided at link time.
- For a better user experience we can choose not to delegate to the user the responsability
  of linking the additional libraries/objects during link-time but rather make Glow capable
  to archive the additional required objects files to the final bundle such that the bundle
  is self contained and the dependency is hidden from the user. During LLVMIRGen we can specify
  to add an extra object file to the bundle by using the function:
  ```
  void LLVMIRGen::addBundleObject(llvm::StringRef objectName);
  ```
  The above function is provided with the object name which is assumed to be present during Glow
  build time in this directory.


More specifically, for our internal use case:
- We currently have some ARM libraries integrated in Glow as source code in LIBJIT which is pre-compiled with Clang to LLVM-IR using the default LIBJIT flow.
- We noted that we get a better performance if we use other compiler (e.g. ARM GCC).
- What we want to do is to pre-compile the library using other compiler to an object file and provide it as dependency to the bundle at link-time (the bundle this time will only call external functions which are not part of LIBJIT but part of the external object file)
- We want this process to be hidden from the user because:
  - We do not want to have an additional step in which we instruct the user to provide the extra object file
  - We have project examples/documentation we do not want to update
  - We don't want the break the current way of using Glow just because the bundle has some external dependencies which Glow could provide